scipy/interpolate/interpolate.py

```diff
@@ -449,7 +446,11 @@ def __init__(self, x, y, kind='linear', axis=-1,
 
         # Interpolation goes internally along the first axis
         self.y = y
-        y = self._reshape_yi(y)
+        self._y = self._reshape_yi(self.y)
+        self.x = x
+        del y, x
```

Comment:
Just makes it clear that the class attributes should be used instead. I'll
add a comment. Or if you prefer it can be removed because functionally the
code is the same without it.